core/controlplane/config/templates/cloud-config-worker

```diff
@@ -24,14 +25,6 @@ INSTANCE_ID=$(curl -s http://169.254.169.254/latest/meta-data/instance-id)
 exec /usr/bin/coreos-cloudinit --from-file /var/run/coreos/$USERDATA_FILE
 {{ end }}
 
-{{ define "instance" -}}
-{ "Fn::Base64": { "Fn::Join" : ["", [
-  "#!/bin/bash -xe\n",
-  {"Fn::Join":["",[ "echo '{{.StackNameEnvVarName}}=", { "Ref": "AWS::StackName" }, "' >> {{.StackNameEnvFileName}}\n" ]]},
-  {{ (execTemplate "instance-script" .) | toJSON  }}
```

Comment:
Just trying to understand the assumption made hre, but probably you could alternatively add the fingerprint here:
```json
{"Fn::Join":["",[  "# s3-part-fingerprint: ", {{ (execTemplate "s3" .) | fingeprint }} ]]},
{"Fn::Join":["",[ "echo '{{.StackNameEnvVarName}}=", { "Ref": "AWS::StackName" }, "' >> {{.StackNameEnvFileName}}\n" ]]},
{{ (execTemplate "instance-script" .) | toJSON  }}
```

so that the brand-new `RenderTemplate` func becomes unnecessary?

Reply:
Hmm, I'm not super expert at templating but if fingerprint is available in the Func map then I don't see why that wouldn't work.
A question why have a different method of generating the node templates from etcd and controller?  One of my thoughts was that they should work the same?

Reply:
Ah, yes, good point. I'm a bit confused 😉 

We definitely should render all the node templates in the same way, for consistency. Your change to introduce `RenderInstance` and the improvement to make worker nodes use that func for userdata generation is definitely a good thing.

Back to current implementation, I believe that the worker template is slightly more advanced than other templates because it achieves the same goal of `cfn-signal` and `cfn-init` with reduced AWS API calls. Hopefully we could keep that.

Reply:
Maybe append this part of the `instance-script` to every "instance" template for all the node types, via `RenderInstance` func?

```
{ "Fn::Base64": { "Fn::Join" : ["", [
"#!/bin/bash -xe\n",
{"Fn::Join":["",[ "echo '{{.StackNameEnvVarName}}=", { "Ref": "AWS::StackName" }, "' >> {{.StackNameEnvFileName}}\n" ]]},
```

Reply:
Hmm - that's an interesting idea!  Let me go away and have a look!